### PR TITLE
Add `shell.shellHook` and `pkgs` options to offchain module

### PR DESCRIPTION
**Pull request by t4ccer**
### Summary of changes

- Added `shell.shellHook` and `pkgs` options in offchain module

Not sure if `pkgs` option is a good idea, but I need to be able to apply overlays to packages passed to `purescriptProject` and `bundlePursProject`.

### Tested on
- [ ] [agora](https://github.com/Liqwid-Labs/agora)
- [ ] [liqwid-libs](https://github.com/Liqwid-Labs/liqwid-libs)
- [ ] [oracle-offchain](https://github.com/Liqwid-Labs/oracle-offchain)
- [x] Other

### CODE REVIEW

**Review by SeungheonOh — APPROVED** [review]
Looks good to me, is this on onchain? If not, it'd be nice to add it on onchain as well

**Comment by t4ccer** [comment]
It's only offchain now, I'm adding these as I need them. I can port it to onchain module tomorrow though

File: nix/offchain.nix [hunk 1]
```diff
@@ -181,6 +202,15 @@ in
                 type = types.path;
               };
 
+              pkgs = lib.mkOption {
+                description = ''
+                  Package set to use. If specified, you must also manually apply
+                  CTL overlays.
```
Comment (emiflake) [inline nix/offchain.nix:208]:
```suggestion
                  CTL overlays.
                  
                  Added in: 2.3.0.
```

**Review by emiflake — APPROVED** [review]
Looks good to me!

**Pull request merged**